inc/domainrecordtype.class.php

```diff
@@ -42,46 +42,225 @@ class DomainRecordType extends CommonDropdown
+               'placeholder' => 'eg. 7200',
+            ],
+            [
+               'label'       => 'Zone expiry timer',
+               'placeholder' => 'eg. 1209600',
+            ],
+            [
+               'label'       => 'Minimum TTL',
+               'placeholder' => 'eg. 300',
+            ],
+         ],
       ], [
          'id'        => 9,
          'name'      => 'SRV',
-         'comment'   => 'Location of service'
+         'comment'   => 'Location of service',
+         'fields'    => [
+            [
+               'label'       => 'Priority',
+               'placeholder' => 'eg. 0',
+            ],
+            [
+               'label'       => 'Weight',
+               'placeholder' => 'eg. 10',
+            ],
+            [
+               'label'       => 'Port',
+               'placeholder' => 'eg. 5060',
+            ],
+            [
+               'label'       => 'Target',
+               'placeholder' => 'eg. sip.example.com',
+            ],
+         ],
       ], [
          'id'        => 10,
          'name'      => 'TXT',
-      'comment'    => 'Descriptive text'
+         'comment'   => 'Descriptive text',
+         'fields'    => [
+            [
+               'label'       => 'TXT record data',
+               'placeholder' => 'Your TXT record data',
+               'quote_value' => true,
+            ],
+         ],
+      ], [
+         'id'        => 11,
+         'name'      => 'CAA',
+         'comment'   => 'Certification Authority Authorization',
+         'fields'    => [
+            [
+               'label'       => 'Flag',
+               'placeholder' => 'eg. 0',
+            ],
+            [
+               'label'       => 'Tag',
+               'placeholder' => 'eg. issue',
+            ],
+            [
+               'label'       => 'Value',
+               'placeholder' => 'eg. letsencrypt.org',
+               'quote_value' => true,
+            ],
+         ],
       ]
    ];
 
+
+   function getAdditionalFields() {
+      return [
+         [
+            'name'  => 'fields',
+            'label' => __('Fields'),
+            'type'  => 'fields',
+         ]
+      ];
+   }
+
+   public function displaySpecificTypeField($ID, $field = []) {
+      $field_name  = $field['name'];
+      $field_type  = $field['type'];
+      $field_value = $this->fields[$field_name];
+
+      switch ($field_type) {
+         case 'fields':
+            $printable = json_encode(json_decode($field_value), JSON_PRETTY_PRINT);
+            echo '<textarea name="' . $field_name . '" cols="75" rows="25">' . $printable . '</textarea >';
+            break;
+      }
+   }
+
+   public function prepareInputForAdd($input) {
+      if (!array_key_exists('fields', $input)) {
+         $input['fields'] = '[]';
+      } else {
+         $input['fields'] = Toolbox::cleanNewLines($input['fields']);
+      }
+
+      if (!$this->validateFieldsDescriptor($input['fields'])) {
```

Comment:
I don't know if we want an enforced validation.

This need to be discussed with customer (on first week of august) and also checked against powerdns integration

Reply:
This is validation of the provided JSON. We have to validate it on input to be sure that it can be used to build the hlper form. Value is set to `[]` by default and so is not mandatory.